### PR TITLE
py-gevent: add version 1.5

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256.

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Dependencies need to be updated too

**Comment by vvolkl** [comment]
@adamjstewart ok I'll check -- I was mostly trying to fix a build failure I encountered with the previous version 1.3. It seemed to work ok now, but I'll take a look at the repo to catch any range changes and optional ones

File: var/spack/repos/builtin/packages/py-gevent/package.py [hunk 1]
```diff
@@ -12,10 +12,15 @@ class PyGevent(PythonPackage):
...
-    depends_on('py-setuptools@24.2:',   type='build')
-    depends_on('py-cython@0.27:',       type='build')
-    depends_on('py-cffi@1.4.0:',        type=('build', 'run'))
-    depends_on('py-greenlet@0.4.13:',   type=('build', 'run'))
+    depends_on('py-setuptools@24.2:',   type='build', when='@:1.4.99')
+    depends_on('py-setuptools@40.8.0:', type='build', when='@1.5.0:')
+    depends_on('py-cython@0.27:',       type='build', when='@:1.4.99')
+    depends_on('py-cython@0.29.14:',    type='build', when='@1.5.0:')
+    depends_on('py-cffi@1.4.0:',        type=('build', 'run'), when='@:1.4.99')
+    depends_on('py-cffi@1.12.3:',        type=('build', 'run'), when='@1.5.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gevent/package.py]:
```suggestion
    depends_on('py-cffi@1.12.2:',        type=('build', 'run'), when='@1.5.0:')
```

File: var/spack/repos/builtin/packages/py-gevent/package.py [hunk 2]
```diff
@@ -12,10 +12,15 @@ class PyGevent(PythonPackage):
...
+    depends_on('py-greenlet@0.4.13:',   type=('build', 'run'), when='@:1.4.99')
+    depends_on('py-greenlet@0.4.14:',   type=('build', 'run'), when='@1.5.0:')
     depends_on('python@2.7:2.8,3.4:',   type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gevent/package.py:26]:
```suggestion
    depends_on('python@2.7:2.8,3.4:',   type=('build', 'run'))
    depends_on('python@2.7:2.8,3.5:',   type=('build', 'run'), when='@1.5:')
```

File: var/spack/repos/builtin/packages/py-gevent/package.py [hunk 3]
```diff
@@ -12,10 +12,15 @@ class PyGevent(PythonPackage):
...
     pypi = "gevent/gevent-1.3a2.tar.gz"
 
+    version('1.5.0', sha256='b2814258e3b3fb32786bb73af271ad31f51e1ac01f33b37426b66cb8491b4c29')
     version('1.3a2', sha256='f7ab82697111ea233c7beeadf5240f669dfad9c4bbc89a3ec80a49e2c48a65bd')
 
-    depends_on('py-setuptools@24.2:',   type='build')
-    depends_on('py-cython@0.27:',       type='build')
-    depends_on('py-cffi@1.4.0:',        type=('build', 'run'))
-    depends_on('py-greenlet@0.4.13:',   type=('build', 'run'))
+    depends_on('py-setuptools@24.2:',   type='build', when='@:1.4.99')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gevent/package.py]:
```suggestion
    depends_on('py-setuptools@24.2:',   type='build', when='@:1.4')
```
Same for others

**Review by adamjstewart — CHANGES_REQUESTED** [review]
It's unclear to me if cffi is actually required on systems other than Windows, but I'm fine with leaving it in there like it already was.

File: var/spack/repos/builtin/packages/py-gevent/package.py [hunk 4]
```diff
@@ -12,10 +12,16 @@ class PyGevent(PythonPackage):
...
-    depends_on('py-cffi@1.4.0:',        type=('build', 'run'))
-    depends_on('py-greenlet@0.4.13:',   type=('build', 'run'))
+    depends_on('py-setuptools@24.2:',   type='build', when='@:1.4')
+    depends_on('py-setuptools@40.8.0:', type='build', when='@1.5.0:')
+    depends_on('py-cython@0.27:',       type='build', when='@:1.4.99')
+    depends_on('py-cython@0.29.14:',    type='build', when='@1.5.0:')
+    depends_on('py-cffi@1.4.0:',        type=('build', 'run'), when='@:1.4.99')
+    depends_on('py-cffi@1.12.2:',        type=('build', 'run'), when='@1.5.0:')
+    depends_on('py-greenlet@0.4.13:',   type=('build', 'run'), when='@:1.4.99')
+    depends_on('py-greenlet@0.4.14:',   type=('build', 'run'), when='@1.5.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-gevent/package.py]:
```suggestion
    depends_on('py-setuptools@24.2:',   type='build', when='@:1.4')
    depends_on('py-setuptools@40.8:', type='build', when='@1.5:')
    depends_on('py-cython@0.27:',       type='build', when='@:1.4')
    depends_on('py-cython@0.29.14:',    type='build', when='@1.5:')
    depends_on('py-cffi@1.4:',        type=('build', 'run'), when='@:1.4')
    depends_on('py-cffi@1.12.2:',        type=('build', 'run'), when='@1.5:')
    depends_on('py-greenlet@0.4.13:',   type=('build', 'run'), when='@:1.4')
    depends_on('py-greenlet@0.4.14:',   type=('build', 'run'), when='@1.5:')
```
Some further simplifications you can make.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**